### PR TITLE
Rewrite PowerShell installation script close #278

### LINKED ISSUE #278: Auto Installer rewrite
@KUTlime 

Also, if possible add solution for these issues.

* #276 
* #274

* All versions of Windows 10 and above should be supported. 
* If possible add support for older versions.

### PR DESCRIPTION
**Pull request by KUTlime**
> This is a rewrite of the PowerShell installation script for Block the Spot.

I refactored the current version of the script to follow PowerShell best practice.
In addition, I implemented two new features tracked in #278.

The first improvement is a fix when the script is executed with administrator privileges. The only possible way how to do it, is use the Windows Task Scheduler. This limits the possible extension to Linux.

The second improvement is a progress bar. I implemented [this method](https://stackoverflow.com/questions/21422364/is-there-any-way-to-monitor-the-progress-of-a-download-using-a-webclient-object/70470467#70470467) which works both in Windows PowerShell and PowerShell. 
The method downloads bytes from a network stream into a buffer. The buffer size effects the overall download speed.

Unfortunately, the method doesn't perform well in Windows PowerShell. Because of that, the script uses `BitsTransfer` module when available. `BitTransfer` delivers both, a reasonable speed and a progress bar.

Benchmarks indicate that use of local implementation for downloading a file with progress bar is 4x times slower than `WebClient`. Hence, its use should be a subject of discussion in this PR.

## Performance on Windows PowerShell

```log
Buffer size 1 MB: 69401.4209 s
Buffer size 2 MB: 67137.3208 s
Buffer size 3 MB: 65460.9411 s                                                                                                                            
Buffer size 4 MB: 65651.293 s                                                                                                                             
Buffer size 5 MB: 66862.3019 s                                                                                                                            
Buffer size 6 MB: 64633.4395 s                                                                                                                            
Buffer size 7 MB: 64618.6908 s                                                                                                                            
Buffer size 8 MB: 66936.5524 s                                                                                                                            
Buffer size 9 MB: 68796.2437 s                                                                                                                            
Buffer size 10 MB: 66043.5965 s                                                                                                                           
Benchmark (Start-BitsTransfer) 22444.7061 s
Benchmark (WebClient) 16687.8525 s
```

## Performance on PowerShell 7.2.1

```log
Buffer size 1 MB: 13460.6372 s
Buffer size 2 MB: 13006.2817 s
Buffer size 3 MB: 19726.1153 s
Buffer size 4 MB: 17741.8723 s
Buffer size 5 MB: 13626.0807 s
Buffer size 6 MB: 19403.2616 s
Buffer size 7 MB: 19278.4926 s
Buffer size 8 MB: 25279.528 s
Buffer size 9 MB: 16269.2725 s
Buffer size 10 MB: 17472.1725 s
Benchmark (Start-BitsTransfer) 23863.6204 s
Benchmark (WebClient) 15484.6174 s
```

## Testing of installation

| PowerShell version | Non-admin | Admin |
|-|-|-|
| v5.1.19041.1320| ✅ | ✅ |
| v7.2.1 | ✅ | ✅ |

## How is a file is downloaded?

| PowerShell version | BitTransfer Available | Download method |
|-|-|-|
| v5.1.19041.1320| ✅ | BitTransfer |
| v7.2.1 | ✅ | Local method |
| v5.1.19041.1320|  ❌ | Local method |
| v7.2.1 | ❌ | Local method |

### CODE REVIEW

File: install.ps1 [hunk 1]
```diff
@@ -1,257 +1,316 @@
...
 '@
 
-write-host @'
-***************** 
+Write-Host @'
+*****************
 Author: @Nuzair46
-***************** 
+*****************
 '@
```
Comment (Nuzair46) [inline install.ps1:85]: Please remove this block.

**Comment by Nuzair46** [comment]
@KUTlime Is this Supported with all versions of Windows 10 and above?

**Review by Nuzair46 — CHANGES_REQUESTED** [review]

**Comment by KUTlime** [comment]
> @KUTlime Is this Supported with all versions of Windows 10 and above?

Yes.

File: install.ps1 (hunk 1 above)
Comment (KUTlime) [reply install.ps1:85]: May I suggest an update of this block: `Authors: @Nuzair46, @KUTlime` 
I think you still deserve the credit. 😉

Comment (Nuzair46) [reply install.ps1:85]: yea sure. go on.

Comment (KUTlime) [reply install.ps1]: Updated.

**Review by Nuzair46 — APPROVED** [review]

**Pull request merged**